### ISSUE TITLE
Card Declined Test Card Not Triggering Error in SDK

### ISSUE THREAD

**Issue by xanderbuck** [Shopify/mobile-buy-sdk-ios#1088]
I am trying to use the "card declined" test card **4000000000000002** found here on [Shopify's list of test cards](https://help.shopify.com/en/manual/payments/shopify-payments/testing-shopify-payments#simulate-failed-transactions)

When I use this test card with the **checkoutCompleteWithCreditCardV2** mutation, I don't receive any checkoutUserErrors or errors indicating the card was declined (even when confirming payment is ready). It should be noted that when I try using this same bad card in the SDK's web flow, it gives me the error message "card declined" as expected.

Since I am getting a successful response back from the **checkoutCompleteWithCreditCardV2** mutation, I then poll for order completion as suggested in the Mobile Buy SDK documentation as such:

```
let retry = Graph.RetryHandler<Storefront.QueryRoot>(endurance: .finite(30)) { (response, error) -> Bool in
    return (response?.node as? Storefront.Checkout)?.order == nil
}
```

but this polls the whole time and order is always nil. 

I've tried using the other cards in the Simulate failed transactions of that website list, and was able to get success with getting error messages back.

**Comment by xanderbuck** [comment]
Any thoughts @dbart01 @vixdug?

**Comment by walteryaron** [comment]
Hi @xanderbuck 
Did you check this:
Before you can create test orders, you need to enable the test mode for Shopify payments.

**Comment by xanderbuck** [comment]
@walteryaron yep, test mode is enabled, and I've been making test orders no problem. Even using the other test credit cards work fine. Its just the "card declined" credit card that doesn't seem to be throwing me an error.
